src/GafferUI/StandardNodeGadget.cpp

```diff
@@ -573,6 +573,14 @@ void StandardNodeGadget::leave( Gadget *gadget )
 
 bool StandardNodeGadget::dragEnter( GadgetPtr gadget, const DragDropEvent &event )
 {
+	// if the sourceGadget is not capable of making a connection anyway, there's
```

Comment:
I'm going to suggest that second-guessing the implementation of `closestDragDestinationProxy()` like this is a bad idea. That method has one responsibility - to tell us if we own a Gadget that could be a proxy for a given DragDropEvent. By effectively copying part of those internals here, we're making ourselves vulnerable to changes in the internal implementation, and breaking encapsulation by assuming we know how that function will operate.

I'm pretty sure this is also masking a bug in `closestDragDestinationProxy()`, whereby it calls `creator->canCreateConnection()` on a potentially null creator.

This isn't performance-critical code, and we're now actually doing the test _twice_ in many cases now, once inside and once outside.